qiime2/sdk/usage.py

```diff
@@ -6,16 +6,38 @@
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
 
+"""
+The Usage module enables generating examples for multiple QIIME 2 interfaces.
+
+The Usage API provides an interface-agnostic way for QIIME 2 plugin developers
+to define examples of how to use their plugin’s actions. This enables the
+programmatic generation of examples for all QIIME 2 interfaces, eliminating the
+need to maintain specific examples for multiple interfaces.
+
+Examples
+--------
+Please see `qiime2.core.testing.examples` for examples.
+"""
+
 import abc
 import re
 import types
+import typing
 
-from qiime2 import sdk, metadata
-
-# TODO: docstrings
+from qiime2 import core, sdk, metadata
 
 
 class UsageAction:
+    """
+    Parameters
+    ----------
+    plugin_id : str
+        Plugin ID.
+    action_id : str
+        Action ID.
+    """
+
+    # TODO If *arg here is necessary, create an example
```

Comment:
I wasn't sure what the type annotation for `*` should be in the constructor here since I couldn't find any examples of it being used.  I only see `UsageAction` being instantiated with the keyword arguments.  I have a feeling like this is a stupid question, but why is `*` a part of the signature for `UsageAction`'s constructor? Also, why isn't it `*args` instead?

Reply:
Great question! Check out [PEP3102](https://www.python.org/dev/peps/pep-3102/) for details, also:
https://python-3-for-scientists.readthedocs.io/en/latest/python3_advanced.html#keyword-only-arguments